### PR TITLE
Add the ability to override log level values

### PR DESCRIPTION
**Pull request by michael-bud**

### CODE REVIEW

**Comment by michael-bud** [comment]
Not sure who the maintainers are anymore but could this be looked at please?

@dgsb @sirupsen @freeformz @freeformzSFDC

**Comment by freeformz** [comment]
@michael-bud Please describe the use case(s) a little so that we can do our best to understand the problem(s) being solved by this change.

**Comment by michael-bud** [comment]
We have a number of different technologies across our architecture which when get fed into a centralised logging system. In make it easier to search for things of 'warn' log level across Go, Python, PHP, Java etc. projects we try to standardise the log level values (e.g. all the same case) but this requires some flexibility in the terms used; similarly we use the functionality that already exists in logrus that allows for renaming of certain fields to standardise those too.

I don't know any of the people who did the +1s on the original PR OP; they may have other use cases too.

File: json_formatter.go [hunk 1]
```diff
@@ -43,6 +54,21 @@ type JSONFormatter struct {
 	// }
 	FieldMap FieldMap
 
+	// LogLevelMap allows users to customize the values used for the various log levels
+	// As an example:
+	// formatter := &JSONFormatter{
+	//   	LogLevelMap: LogLevelMap{
+	// 		 PanicLevel:  1,
```
Comment (freeformz) [inline json_formatter.go]: This example isn't valid. The value on the right side needs to be a string, so "1", "2", "3", ...

**Comment by michael-bud** [comment]
(a) Yep, as you've pointed out, it's only really possible by replacing the whole JSON formatter with a custom one
(b) The reason I didn't do it as a custom function was to keep consistency with where we're doing this elsewhere in the JSON formatter, specifically where we allow customisation of the field names for things like the message, log level and timestamp.

File: json_formatter.go [hunk 2]
```diff
@@ -20,6 +20,17 @@ func (f FieldMap) resolve(key fieldKey) string {
 	return string(key)
 }
 
+// LogLevel map allows customization of the values used to describe log levels
+type LogLevelMap map[Level]string
```
Comment (freeformz) [inline json_formatter.go]: Please rename to `LevelMap` and fix the good to correspond.

File: json_formatter.go (hunk 1 above)
Comment (freeformz) [inline json_formatter.go]: Please make the example valid. Better yet, make an `ExampleLevelMap` function in the [test](https://golang.org/pkg/testing/#hdr-Examples) file.

**Comment by freeformz** [comment]
@michael-bud That makes sense. I requested a few (minor) changes that I'd like to see before merging.

**Comment by freeformz** [comment]
@michael-bud So...
After I posted that, and was reading #463, I started to wonder if this is solely a function of the JSONFormatter or not. So now I'm not so sure about my statements above.

For instance Level currently hard defines the string definition used for the stock levels. Maybe that should be more flexible and could help finish off #463. Thoughts ?

**Comment by michael-bud** [comment]
@freeformz I'd rather not try and work that into this as I think it will make this significantly more complex than it needs to be and #463 looks like quite a massive change whereas this is relatively minor. The use cases are also quite different.

**Comment by lnenad** [comment]
+1, i think having the ability to define custom levels is a necessary thing.

**Comment by michael-bud** [comment]
@freeformz This PR is waiting for a response from you I think?

**Comment by iamolegga** [comment]
Hi, is this ready to be merged? I guess there are just minor fixes left, really looking forward for this feature 👍

**Comment by zx8** [comment]
Would really love to see this merged, but progress seems to have stalled.

I'm more than happy to pick it up and make the changes necessary to get it merged, if @michael-bud / @freeformz can confirm the current state of things?

**Pull request closed without merging**